tests/auto-merge.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { beforeEach, describe, expect, test, vi } from "vitest";
+
+import autoMerge, { NotDependabotPrError } from "../src/auto-merge";
+import * as github from "../src/lib/github";
+import { contextExample } from "./context";
+
+vi.mock("../src/lib/github", () => ({
+  comment: async () => undefined,
+  review: async () => undefined,
+}));
+
+describe("#autoMerge", () => {
+  const commentSpy = vi.spyOn(github, "comment");
+  const reviewSpy = vi.spyOn(github, "review");
```

Comment:
I think you could just `vi.mock("../src/lib/github")` and then use `vi.mocked(github.comment)`